cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/IndexManagerImpl.java

```diff
@@ -94,26 +95,17 @@
      *  Constructs a new IndexManager which indexes documents in 'datastore'
      *  @param database The {@link Database} to index
      */
-    public IndexManagerImpl(Database database) {
+    public IndexManagerImpl(Database database) throws IOException, SQLException {
         this.database = database;
         validFieldName = Pattern.compile(INDEX_FIELD_NAME_PATTERN);
 
-        final String filename = ((DatabaseImpl) database).extensionDataFolder(EXTENSION_NAME) + File.separator
-                                                                              + "indexes.sqlite";
-        final KeyProvider keyProvider = ((DatabaseImpl) database).getKeyProvider();
-
-        SQLDatabaseQueue queue = null;
+        final File file = new File(((DatabaseImpl) database).extensionDataFolder(EXTENSION_NAME), "indexes.sqlite");
```

Comment:
Thinking about this some more, I don't think we should couple this to the implementation of the `Database` perhaps we should have a package level interface which exposes the extensions file path, and that is what gets  passed into this method.